exchanges/huobi/huobi_test.go

```diff
@@ -2870,74 +2851,52 @@ func TestUpdateTickers(t *testing.T) {
 	t.Parallel()
 	for _, a := range h.GetAssetTypes(false) {
 		err := h.UpdateTickers(context.Background(), a)
-		assert.NoErrorf(t, err, "asset %s", a)
-
+		require.NoErrorf(t, err, "asset %s", a)
 		avail, err := h.GetAvailablePairs(a)
 		require.NoError(t, err)
-		for x := range avail {
-			_, err = ticker.GetTicker(h.Name, avail[x], a)
-			assert.NoError(t, err)
+		for _, p := range avail {
+			_, err = ticker.GetTicker(h.Name, p, a)
+			assert.NoErrorf(t, err, "Could not get ticker for %s %s", a, p)
 		}
 	}
 }
 
-func TestConvertContractShortHandToExpiry(t *testing.T) {
+func TestPairFromContractExpiryCode(t *testing.T) {
 	t.Parallel()
-	tt := time.Now()
-	cp := currency.NewPair(currency.BTC, currency.NewCode("CW"))
-	cp, err := h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.NotEqual(t, "CW", cp.Quote.String())
-	tick, err := h.FetchTicker(context.Background(), cp, asset.Futures)
-	if assert.NoError(t, err) {
-		assert.NotZero(t, tick.Close)
-	}
-
-	cp = currency.NewPair(currency.BTC, currency.NewCode("NW"))
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.NotEqual(t, "NW", cp.Quote.String())
-	tick, err = h.FetchTicker(context.Background(), cp, asset.Futures)
-	if assert.NoError(t, err) {
-		assert.NotZero(t, tick.Close)
-	}
-
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CQ"))
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.NotEqual(t, "CQ", cp.Quote.String())
-	tick, err = h.FetchTicker(context.Background(), cp, asset.Futures)
-	if assert.NoError(t, err) {
-		assert.NotZero(t, tick.Close)
-	}
-
-	// calculate a specific date
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CQ"))
-	tt = time.Date(2021, 6, 3, 0, 0, 0, 0, time.UTC)
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.Equal(t, "210625", cp.Quote.String())
 
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CW"))
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.Equal(t, "210604", cp.Quote.String())
+	h := new(HUOBI) //nolint:govet // Intentional shadow to avoid future copy/paste mistakes
+	require.NoError(t, testexch.Setup(h), "Test Instance Setup must not fail")
 
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CWif hat"))
-	_, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.ErrorIs(t, err, errInvalidContractType)
+	_, err := h.FetchTradablePairs(context.Background(), asset.Futures)
```

Comment:
Since `UpdateTradablePairs` is called in `TestMain` you could assign it to your shadow h, or use your helper `UpdatePairsOnce`

Reply:
Could do. However for this test-case I wanted to avoid any indirectness, because we expect `FetchTradablePairs` to be doing the thing itself, and now.